lib/action_view/component/render_monkey_patch.rb

```diff
@@ -15,15 +15,23 @@ def render(options = {}, args = {}, &block)
 
           options.render_in(self, &block)
         elsif options.is_a?(Class) && options < ActionView::Component::Base
-          options.new(args).render_in(self, &block)
+          new_component(options, args).render_in(self, &block)
         elsif options.is_a?(Hash) && options.has_key?(:component)
-          options[:component].new(options[:locals]).render_in(self, &block)
+          new_component(options[:component], options[:locals]).render_in(self, &block)
         elsif options.respond_to?(:to_component_class) && !options.to_component_class.nil?
-          options.to_component_class.new(options).render_in(self, &block)
+          new_component(options.to_component_class, options).render_in(self, &block)
         else
           super
         end
       end
+
+      private
+
+      def new_component(component_class, options)
+        arity = component_class.instance_method(:initialize).arity
+
+        (options.blank? && arity.zero?) ? component_class.new : component_class.new(options)
+      end
```

Comment:
I'm not sure how I feel about my use of `#blank?` and `#zero?` here. If the user is using the `#to_component_class` style of `render` (`render(@post)`) and their object  implements either of those methods, then they might see some confusing results.

That said, I want to inspect the options coming in, as to avoid dropping arguments incorrectly. I think that if the user passes options to a 0-arity initializer, they _should_ see an ArgumentError to help debug.

Reply:
My style would be to add a code comment around this that details the logic here. As you point out in the PR comment above we're trying to preserve the ArgumentError etc. The tests cover this case as some form of documentation but it might be worth spelling it out here. 

Another though how about we call this method `safe_new_component` to hint at what it's trying to do?

Reply:
OK, I made the suggestions you recommended.